lookup.go

```diff
@@ -157,29 +157,33 @@ func doOutput(out <-chan string, path string, wg *sync.WaitGroup) error {
 }
 
 // read input file and put results into channel
-func doInput(in chan<- interface{}, path string, wg *sync.WaitGroup, zonefileInput bool) error {
-	var f *os.File
-	if path == "" || path == "-" {
-		f = os.Stdin
+func doInput(in chan<- interface{}, path string, wg *sync.WaitGroup, zonefileInput bool, passedName string) error {
+	if len(passedName) > 0 {
```

Comment:
Rather than a big `if...else` I'd probably put a `defer func() { close(in) }()` at the top, and then just have the `if len(passedName) > 0` block return.